cardano-db/src/Cardano/Db/Schema.hs

```diff
@@ -160,6 +160,22 @@ share
     deriving Eq
     deriving Show
 
+  -- A table with all the different types of total balances
+  -- This is only populated when we are close to the tip of the chain and only
+  -- after epoch boundaries.
+  AdaPots
+    blockId             BlockId             OnDeleteCascade
+    slotNo              Word64              sqltype=uinteger
```

Comment:
This should probably have a `DeleteOnCascade` on it. I also wonder if we should not use the epoch number there instead of the slot number.

Reply:
`blockId` which is the foreign key, does have a `OnDeleteCascade` already.
Also from the blockId we can always find the block and get all the information (epoch or slot). I just replicate the slot here to make it easier to find the latest entry of AdaPots without joins.

Reply:
Brain fart! Somehow I missed the `OnDeleteCascade` on the `blockId` .

The value in this `AdaPot` should be constant for a whole epoch right? If that is the case, this field should probably be `epochNo` instead of `slotNo`. Is there any relevant different in the query between `epochNo` and `slotNo`?

Reply:
> The values in this AdaPot should be constant for a whole epoch right?
It seems the reserve and treasury are constant, but the utxo changes for every block.

Reply:
Is it worth dropping the UTxO  component for this table? Would that still be useful?